File: src/lib/api/debugClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class Debug {
  constructor(private readonly host: RPCHost) {}

  public pProf(arg: pb.IPProfRequest = new pb.PProfRequest()): Promise<pb.PProfResponse> {
    return this.host.expectOne(this.host.call("Debug/PProf", new pb.PProfRequest(arg)));
  }
  public readMetrics(
    arg: pb.IReadMetricsRequest = new pb.ReadMetricsRequest()
  ): Promise<pb.ReadMetricsResponse> {
    return this.host.expectOne(this.host.call("Debug/ReadMetrics", new pb.ReadMetricsRequest(arg)));
  }
}

File: android/app/src/main/java/gg/strims/ppspp/rpc/DebugClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class DebugClient(filepath: String) : RPCClient(filepath) {

    suspend fun pProf(
        arg: PProfRequest = PProfRequest()
    ): PProfResponse =
        this.callUnary("Debug/PProf", arg)

    suspend fun readMetrics(
        arg: ReadMetricsRequest = ReadMetricsRequest()
    ): ReadMetricsResponse =
        this.callUnary("Debug/ReadMetrics", arg)

}
